src/test/codegen/sse42-implies-crc32.rs

```diff
@@ -0,0 +1,17 @@
+// min-llvm-version: 14.0
+// needs-llvm-components: x86
+// compile-flags: --target=x86_64-unknown-linux-gnu -Copt-level=3
+
+#![crate_type = "lib"]
+
+#[cfg(target_arch = "x86_64")]
+#[target_feature(enable = "sse4.2")]
+#[no_mangle]
+pub unsafe fn crc32sse(v: u8) -> u32 {
+    use std::arch::x86_64::*;
+    let mut out = !0u32;
+    _mm_crc32_u8(out, v);
```

Comment:
I think you want to do the same with `crc32` return value here as with the one in the assembly test. Otherwise the intrinsic will get optimized out as we've seen and doesn't actually test much at all.

(I see that in this instance we test the attribute, but even if the attribute is present, its plausible that something else might break down the line much the same way as the breakage this PR is fixing)

Reply:
Good catch! Fixed.